### PR TITLE
[lexical-rich-text] Bug Fix: HeadingNode.insertNewAfter

### PR DESCRIPTION
**Pull request by GermanJablo**
## Description

If a paragraph is inserted when the selection is at the end of a heading, a paragraph should be inserted.

However, the logic for detecting whether the selection was at the end of the heading was incorrect. It did not work if, for example, the heading had multiple TextNodes.

### CODE REVIEW

**Comment by GermanJablo** [comment]
Before and After: https://www.loom.com/share/522f8f3bd808464bb07c7019c3abdf36

**Comment by potatowagon** [comment]
i can reproduce the issue, had to create multiple text nodes in heading, eg. bold "Hello" in "Hello world"

File: packages/lexical-rich-text/src/__tests__/unit/LexicalHeadingNode.test.ts [hunk 1]
```diff
@@ -106,6 +106,56 @@ describe('LexicalHeadingNode tests', () => {
...
+
+    test('HeadingNode.insertNewAfter() end', async () => {
+      const {editor} = testEnv;
+      let headingNode: HeadingNode;
+      await editor.update(() => {
+        const root = $getRoot();
+        headingNode = new HeadingNode('h1');
+        const headingTextNode1 = $createTextNode('hello');
+        const headingTextNode2 = $createTextNode(' world');
+        headingTextNode2.setFormat('bold');
```
Comment (potatowagon) [inline packages/lexical-rich-text/src/__tests__/unit/LexicalHeadingNode.test.ts:141]: 👍

**Review by potatowagon — APPROVED** [review]
thanks for the fix!

**Pull request merged**